From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 9 Jan 2024 12:18:24 -0500
Subject: [PATCH] cmake: Make global C/C++ std a cache variable so it can be
 overridden

The standard can be set from the command line when running cmake by
passing e.g. "-DCMAKE_CXX_STANDARD=17".

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 338df43c7..36ff72e52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -451,9 +451,9 @@ endif()
 # Set minimal C and C++ standards
 ################################################################################
 if(NOT (CMAKE_VERSION VERSION_LESS "3.1"))
-    set(CMAKE_C_STANDARD 11)
-    set(CMAKE_CXX_STANDARD 14)
-    set(CMAKE_CXX_EXTENSIONS OFF)
+    set(CMAKE_C_STANDARD 11 CACHE STRING "Default minimum C standard")
+    set(CMAKE_CXX_STANDARD 14 CACHE STRING "Default minimum C++ standard")
+    set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Default enable/disable compiler extensions")
 else()
     add_compile_options("$<$<STREQUAL:$<TARGET_PROPERTY:LINKER_LANGUAGE>,C>:-std=gnu11>")
     set(CMAKE_C_STANDARD 11)   # set variable just for reporting
-- 
2.42.0

